client/internal/engine.go

```diff
@@ -857,3 +863,116 @@ func (e *Engine) receiveSignalEvents() {
 
 	e.signal.WaitStreamConnected()
 }
+
+func (e* Engine) parseNATExternalIPMappings() []string {
+	var mappedIPs []string
+	for _, mapping := range e.config.NATExternalIPs {
+		var external, internal string
+		var externalIP, internalIP net.IP
+		var err error
+		split := strings.Split(mapping, "/")
+		if len(split) > 1 {
+			internal = split[1]
+			internalIP = net.ParseIP(internal)
+			if internalIP == nil {
+				// not a properly formattted IP address, maybe it's interface name?
+				internalIP, err = findIPFromInterfaceName(internal)
+				if err != nil {
+					log.Warnf("error finding interface IP for interface '%s', ignoring external mapping '%s': %v", internal, mapping, err)
+					break
+				}
+			}
+		}
+		external = split[0]
+		if external == "stun" {
+			externalIP, err = e.findNATExternalIPFromSTUN(internalIP)
+			if err != nil {
+				log.Warnf("error finding external IP via stun, ignoring external IP mapping '%s': %v", mapping, err)
+				break
+			}
+		} else {
+			externalIP = net.ParseIP(external)
+			if externalIP == nil {
+				log.Warnf("invalid external IP, ignoring external IP mapping '%s'", mapping)
+				break
```

Comment:
we should return an empty list to avoid half parsed values and hard to troubleshoot cases

Reply:
The result is initialized as empty list and only parsable mappings are appended to returned result.  It's possible I don't understand the comment.

Reply:
the comment is more toward making the parseNATExternalIPMappings function return either a valid result list by parsing all the elements from e.config.NATExternalIPs or returns an empty list in case one element fails to parse.